### PR TITLE
Feat/update create script granular perms

### PR DESCRIPTION
**Pull request by love-bhardwaj**
## What this PR does?

- Add script to create an app using data from a YAML file (kind of apps manifest)
- Move common functionality between the two create scripts into a common file
- Update docs to reflect the changes

### CODE REVIEW

File: src/scripts/create-app.ts [hunk 1]
```diff
@@ -33,37 +20,15 @@ const args = yargs(process.argv.slice(2)).options({
...
+  name: args.name,
+  orgId: args.orgId,
+  redirectUris: args.redirectUris || ['http://localhost:3000/callback'],
+  scopes: args.scopes,
+  authToken: args.authToken,
+})
   .then(handleResult)
-  .catch((err) => console.log(`Error creating app: "${err}", details: "${JSON.stringify(err.response.data)}"`));
+  .catch((err) => {
+    logger('error: ', err);
```
Comment (DarrellMozingo) [inline src/scripts/create-app.ts:32]: What does the `logger` add over the `console.log`?

File: README.md [hunk 2]
```diff
@@ -25,26 +25,28 @@ Also important to mention that we are using [passportjs](https://www.passportjs.
...
+$ npm run create-app -- --authToken=some-token --orgId=some-snyk-org-id --redirect-uris=https://your-domain/callback --scopes=org.read org.project.read --name=test-snyk-app
 ```
 
 (note the extra `--` between `create-app` and the parameters)
 
-- `authToken`(**Required**/**String**): your personal Snyk auth token, obtained from [your account settings page](https://app.snyk.io/account)
+- `authToken` (**Required**/**String**): your personal Snyk auth token, obtained from [your account settings page](https://app.snyk.io/account)
 - `orgId` (**Required**/**String**): the organization id that you want to own the Snyk App (obtained by clicking the cog in the upper right corner of the Snyk console)
 - `redirectUris` (**Optional**/**String Array**): a space separated list of redirect uris for your app, defaults to `http://localhost:3000/callback` when no input provided
 - `scopes` (**Required**/**String Array**): a space separated list of scopes you want your App to be able to request at install time (see [Snyk Apps docs](https://docs.snyk.io/integrations/snyk-apps) for allowed values)
```
Comment (DarrellMozingo) [inline README.md:52]: Should we note these are permissions here? I'm not sure users will have ever seen them before, or if they're even publicly documented?

File: README.md [hunk 3]
```diff
@@ -58,6 +60,40 @@ This will register your new app with Snyk and create the `.env` file (see below)
 - `SCOPES`: the space-separated list of scopes for your Snyk App
 - `ENCRYPTION_SECRET`: secret encryption key used by the demo app to encrypt sensitive data
 
+### Using file
+
+```shell
+$ npm run create-app:from-file -- --authToken=$token
```
Comment (DarrellMozingo) [inline README.md:66]: Could/should we combine the scripts? `--file` causes it to use that, otherwise it expected all the args?

File: README.md [hunk 4]
```diff
@@ -58,6 +60,40 @@ This will register your new app with Snyk and create the `.env` file (see below)
 - `SCOPES`: the space-separated list of scopes for your Snyk App
 - `ENCRYPTION_SECRET`: secret encryption key used by the demo app to encrypt sensitive data
 
+### Using file
```
Comment (DarrellMozingo) [inline README.md:63]: Is this worth adding now? I thought once we land support in the CLI we'd be removing this functionality completely from the demo app and directing them to the CLI?

If we go that route, adding file/manifest support into the CLI might be a better route.

**Review by DarrellMozingo — APPROVED** [review]
Looks good as is! A few questions below on the file support :)

File: src/scripts/create-app.ts (hunk 1 above)
Comment (love-bhardwaj) [reply src/scripts/create-app.ts:32]: Not much but it helps testing it locally and you can control if the logs get printed or not using an env var. Mainly helpful for dev purposes but sometimes the error is not a response error or HTTP error. In those cases `err.response.data` in the `console.log` itself throws an error as the error is `undefined`.

File: README.md (hunk 2 above)
Comment (love-bhardwaj) [reply README.md:52]: Yes, we can do it. I was thinking that we merge this once we get the docs out about permissions.

File: README.md (hunk 4 above)
Comment (love-bhardwaj) [reply README.md:63]: Yes, that would be the intended route. We would want the file argument for the CLI as the permissions list could be large. The only problem I see is the user having to authenticate the Snyk CLI before using it. Using the scripts here they can pass the auth token and it works in one command. Which I feel has an edge, when we want the user to be able to run the demo app right away. Happy to go either way though.

**Pull request closed without merging**